xdp-dump/xdpdump.c

```diff
@@ -1415,8 +1419,14 @@ static bool load_and_attach_trace(struct dumpopt *cfg,
 		trace_link_fentry = bpf_program__attach_trace(trace_prog_fentry);
 		err = libbpf_get_error(trace_link_fentry);
 		if (err) {
-			pr_warn("ERROR: Can't attach XDP trace fentry function: %s\n",
-				strerror(-err));
+			if (err == -ENOTSUPP)
+				pr_warn("ERROR: Can't attach XDP trace fentry "
+					"function due to missing architectural "
+					"support!");
```

Comment:
Could we make the error message a bit more concrete so users understand what's going on? 

We could detect the arch and do something like: `pr_warn("The kernel does not support fentry function attach %s", arch==x86 ? "because it is too old" : "on the current CPU architecture");` ?

Reply:
Will do